.github/workflows/ci-pipeline.yml

```diff
@@ -21,3 +21,34 @@ jobs:
 
       - name: Run tests
         run: docker run todo-app:test
+
+  container_build_and_push:
+    name: Container build and push
+    needs: build_and_test
+    if: github.event_name == 'push' && endsWith(github.ref, 'module9')

```

Comment:
Ooh nice use of `endsWith` - cleaner than most direct equalities!
(And don't forget to update when you merge as in your PR comment 😄 )